backend-rust/src/connection.rs

```diff
@@ -100,24 +114,27 @@ impl<A> ConnectionQuery<A> {
 
         let from = if let Some(a) = after {
             a.parse::<A>().map_err(|e| e.into())?
-        } else {
+        } else if ascending_default_order {
             A::MIN
+        } else {
+            A::MAX
         };
 
         let to = if let Some(b) = before {
             b.parse::<A>().map_err(|e| e.into())?
-        } else {
+        } else if ascending_default_order {
             A::MAX
+        } else {
+            A::MIN
         };
 
         let limit =
             first.or(last).map_or(connection_limit, |limit| connection_limit.min(limit)) as i64;
-
         Ok(Self {
             from,
             to,
             limit,
-            desc: last.is_some(),
+            desc: !(last.is_some() ^ ascending_default_order),
```

Comment:
It seems to me that this would be changing the meaning of this variable, which would break the other places it is used. The new meaning fits better with the name, but it is probably better to rename it to better fit what it means instead of changing it everywhere.